File: src/Entity/Browser/Container/Page/Navbar/Update.php
<?php

declare(strict_types=1);

namespace Yggverse\Yoda\Entity\Browser\Container\Page\Navbar;

use \GtkButton;

use \Yggverse\Yoda\Abstract\Entity\Browser\Container\Page\Navbar\Button;

class Update extends Button
{
    public const IMAGE = 'view-refresh-symbolic';
    public const LABEL = 'Go';
    public const TOOLTIP = 'Update';

    protected function _onCLick(
        GtkButton $entity
    ): void
    {
        $this->navbar->page->update();
    }

    public function refresh(): void
    {
        $this->gtk->set_sensitive(
            !empty(
                $this->navbar->request->getValue()
            )
        );
    }
}

File: src/app/browser/main/tab/page/navbar/update.cpp
#include "update.hpp"

using namespace app::browser::main::tab::page::navbar;

Update::Update()
{
    set_action_name(
        "page.update"
    );

    set_icon_name(
        "view-refresh-symbolic"
    );

    set_tooltip_text(
        _("Update")
    );
}